src/patterns/builder/service-example/example/example.exec.ts

```diff
@@ -0,0 +1,23 @@
+import {backOfficeNetworkService, mobileNetworkService, webNetworkService} from './example.impl'
+
+console.log('Public WEB', {
+  // 'TS2339: Property userService does not exist on type { editingService: EditingService; } & { paymentService: PaymentService; }',
+  // adminUser: webNetworkService.userService.getUser('admin@gmail.com'),
+  editingHistory: webNetworkService.editingService.getEditingHistory(),
+  invoices: mobileNetworkService.paymentService.getInvoices(),
+});
+
+console.log('Mobile', {
+  // 'TS2339: Property userService does not exist on type { editingService: EditingService; } & { paymentService: PaymentService; }',
+  // adminUser: webNetworkService.userService.getUser('admin@gmail.com'),
+  editingHistory: mobileNetworkService.editingService.getEditingHistory(),
+  invoices: mobileNetworkService.paymentService.getInvoices(),
+});
+
+console.log('BackOffice WEB', {
+  adminUser: backOfficeNetworkService.userService.getUser('admin@gmail.com'),
+  // 'TS2339: Property editingService does not exist on type { userService: UserService; }',
+  editingHistory: backOfficeNetworkService.editingService?.getEditingHistory(),
+  // 'TS2339: Property paymentService does not exist on type { userService: UserService; }',
+  invoices: backOfficeNetworkService.paymentService?.getInvoices(),
```

Comment:
I'd like to discuss the differences with the initial constructor argument:
![image](https://github.com/user-attachments/assets/dc07ae42-8f59-47a7-90f6-c936790b1e93)

Reply:
I think that adding default type for Added should fix the issue

```
export class NetworkLayer<Added extends INetworkLayer = {}> implements Builder
```